src/conf_mode/firewall.py

```diff
@@ -244,6 +280,10 @@ def verify_rule(firewall, family, hook, priority, rule_id, rule_conf):
         if {'match_frag', 'match_non_frag'} <= set(rule_conf['fragment']):
             raise ConfigError('Cannot specify both "match-frag" and "match-non-frag"')
 
+    is_node_empty_result = is_node_empty(rule_conf)
+    if is_node_empty_result[0]:
+        raise ConfigError(f'Configuration node {" ".join(is_node_empty_result[1])} may not be empty')
```

Comment:
Please use a tupel here and avoid working with indexes. Also the underscores in e.g. `add_address_to_group` and `connection_status` should be replaced with a hyphen.

```suggestion
    node_empty, node_name = is_node_empty(rule_conf)
    if node_empty:
        tmp = ' '.join(node_name).replace('_', '-')
        raise ConfigError(f'Configuration node {} may not be empty')
```

Reply:
Done!